### PR TITLE
Catch ogr http errors and item not existing

### PR DESCRIPTION
**Pull request by francbartoli**
Catch ogr http errors and item not existing

### CODE REVIEW

**Comment by tomkralidis** [comment]
Looks good @francbartoli . Comments:

- should we rename the class to `ProviderItemNotFound` (will help for support of other standards (i.e. OGC API - Records)?
- should we update the various `get` functions of the various other providers (if they support it)?

**Comment by francbartoli** [comment]
Should it be `ProviderItemNotFoundError` or `ProviderItemNotFound`? The first would be aligned with other names

**Comment by tomkralidis** [comment]
+1

File: pygeoapi/api.py [hunk 1]
```diff
@@ -868,6 +868,13 @@ def get_collection_item(self, headers_, format_, dataset, identifier):
         try:
             LOGGER.debug('Fetching id {}'.format(identifier))
             content = p.get(identifier)
+        except ProviderItemNotFoundError:
```
Comment (tomkralidis) [inline pygeoapi/api.py:871]: Should this error go **after** the `ProviderConnectionError` below for logical sequence?

**Comment by justb4** [comment]
Running tests in Docker take a long time I have been noticing, could be due to networking/DNS issues...

**Pull request merged**